### PR TITLE
windows: make more windows tests pass

### LINKED ISSUE #6820: respect Windows "Hidden" file attribute before it is too late.
### What is the problem this feature would solve?

bun on windows uses libuv for I/O operations and libuv itself doesn't respect Windows "Hidden" file at the moment.

libuv (https://github.com/libuv/libuv/pull/3380).

nodejs (https://github.com/nodejs/node/issues/41093).

I would like bun to take it into consideration before it is too late.

### What is the feature you are proposing to solve the problem?

respect Windows "Hidden" file attribute.

### What alternatives have you considered?

_No response_

### PR DESCRIPTION
**Pull request by paperclover**
some changes to moving spawn({ cmd: [] }) to spawn([], {}) are done because our workspaces type definitions are broken :(

making use our `open` in more places than before because i think our synchronous open is better than libuv's

Fixes #6820

### CODE REVIEW

File: src/bun.js/RuntimeTranspilerCache.zig [hunk 1]
```diff
@@ -543,8 +543,9 @@ pub const RuntimeTranspilerCache = struct {
 
         const cache_dir_fd = brk: {
             if (std.fs.path.dirname(cache_file_path)) |dirname| {
-                const dir = try std.fs.cwd().makeOpenPath(dirname, .{ .access_sub_paths = true });
-                break :brk bun.toLibUVOwnedFD(dir.fd);
+                var dir = try std.fs.cwd().makeOpenPath(dirname, .{ .access_sub_paths = true });
```
Comment (jdalton) [inline src/bun.js/RuntimeTranspilerCache.zig:546]: I wonder if deep down in the implementation if `std.fs.cwd` which uses `std.os.windows.peb()` and `std.os.getcwd` which uses `std.os.windows.GetCurrentDirectoryW` (by way of `kernel32.GetCurrentDirectoryW`) are similarly resolved.

File: src/bun.js/node/node_fs.zig [hunk 2]
```diff
@@ -5187,6 +5198,7 @@ pub const NodeFS = struct {
             _ = Syscall.setFileOffset(fd, args.offset);
         }
         // For certain files, the size might be 0 but the file might still have contents.
+        // https://github.com/oven-sh/bun/issues/1220
```
Comment (jdalton) [inline src/bun.js/node/node_fs.zig:5225]: Wow. Good info.

File: src/cli/install_completions_command.zig [hunk 3]
```diff
@@ -140,13 +140,13 @@ pub const InstallCompletionsCommand = struct {
...
         // This uninstaller file is only written if the current exe is within a path
-        // like `\bun\bin\<whatever>.exe` so that it probably only runs when the
+        // like `bun\bin\<whatever>.exe` so that it probably only runs when the
         // powershell `install.ps1` was used to install.
 
         const image_path = bun.windows.exePathW();
         const image_dirname = image_path[0..(std.mem.lastIndexOfScalar(u16, image_path, '\\') orelse unreachable)];
 
-        if (!std.mem.endsWith(u16, image_dirname, comptime bun.strings.literal(u16, "\\bun\\bin")))
+        if (!std.mem.endsWith(u16, image_dirname, comptime bun.strings.literal(u16, "bun\\bin")))
```
Comment (paperclover) [inline src/cli/install_completions_command.zig:149]: `.bun\bin` but also handle some form of user-install where it is organized in a folder named bun

File: src/bun.js/RuntimeTranspilerCache.zig [hunk 4]
```diff
@@ -543,8 +543,9 @@ pub const RuntimeTranspilerCache = struct {
...
+                errdefer dir.close();
```
Comment (dylan-conway) [inline src/bun.js/RuntimeTranspilerCache.zig:547]: nice

File: src/tmp.zig [hunk 5]
```diff
@@ -28,7 +28,10 @@ pub const Tmpfile = struct {
             if (comptime allow_tmpfile) {
                 switch (bun.sys.openat(destination_dir, ".", O.WRONLY | O.TMPFILE | O.CLOEXEC, perm)) {
                     .result => |fd| {
-                        tmpfile.fd = bun.toLibUVOwnedFD(fd);
+                        tmpfile.fd = switch (bun.sys.makeLibUVOwnedFD(fd, .open, .close_on_fail)) {
+                            .result => |owned_fd| owned_fd,
```
Comment (dylan-conway) [inline src/tmp.zig:32]:
```suggestion
                        tmpfile.fd = switch (bun.sys.toLibUVOwnedFD(fd, .open, .close_on_fail)) {
```

**Review by dylan-conway — APPROVED** [review]
Looks good. There's a build failure but the code is unreachable

**Comment by Jarred-Sumner** [comment]
What tests does this fix?

File: src/js/internal/fs/cp.ts [hunk 6]
```diff
@@ -17,7 +17,6 @@
 const { chmod, copyFile, lstat, mkdir, opendir, readlink, stat, symlink, unlink, utimes } = require("node:fs/promises");
 const { dirname, isAbsolute, join, parse, resolve, sep } = require("node:path");
 
-const SafePromiseAll = Promise.all;
 const PromisePrototypeThen = Promise.prototype.then;
```
Comment (Jarred-Sumner) [inline src/js/internal/fs/cp.ts:20]: The hilarious thing about this is `.$then` exists and so it's less safe than the actual

**Pull request merged**